src/bitcoin-cli.cpp

```diff
@@ -340,26 +372,15 @@ int CommandLineRPC(int argc, char *argv[])
         if (args.size() < 1)
             throw std::runtime_error("too few parameters (need at least command)");
         std::string strMethod = args[0];
-        auto params =
-            RPCConvertValues(strMethod, std::vector<std::string>(args.begin()+1, args.end()));
-
-        if (!params.has_value()) {
-            params.map_error([&](auto failure) {
-                throw std::runtime_error(FormatConversionFailure(strMethod, failure));
-            });
-        }
 
         // Execute and handle connection failures with -rpcwait
         const bool fWait = GetBoolArg("-rpcwait", false);
         do {
             try {
-                const UniValue reply = CallRPC(strMethod, params.value());
-
-                // Parse reply
-                const UniValue& result = find_value(reply, "result");
-                const UniValue& error  = find_value(reply, "error");
+                auto reply = CallRPC(strMethod, std::vector<std::string>(args.begin()+1, args.end()));
```

Comment:
This now repeatedly converts the RPC parameters every second if `-rpcwait` is provided and the RPC is busy. I don't fully understand why `CallRPC` was refactored in this way?

Reply:
Ah, I’ll extract out the conversion. But `CallRPC` was refactored because now we call it again to get the help message if there was a client-side failure. But … help messages don’t need to do any parameter conversion, so perhaps I can extract it without adding duplication.

Reply:
Ok, I extracted out the parameter conversion, so it only happens once.